src/main/java/io/cdap/e2e/pages/locators/CdfNameSpaceAdminLocators.java

```diff
@@ -0,0 +1,160 @@
+  public static WebElement clickOnFinishButton;
+
+  public static WebElement locateKeyProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//input[@placeholder='key']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateValueProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//input[@placeholder='value']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateAddRowButtonProperty(int row) {
+    String xpath = "//*[@data-cy= 'key-value-pair-" + row + "']//span//button[@type='submit' and " +
+        "@class='btn add-row-btn btn-link']";
+    return SeleniumDriver.getDriver().findElement(By.xpath(xpath));
+  }
+
+  public static WebElement locateProvisionerInList(String provisionerName) {
+    return SeleniumDriver.getDriver()
+        .findElement(By.xpath("//div[@data-cy='" + provisionerName + "']"));
+  }
+
+  public static WebElement locateButtonType(String buttonType) {
+    return SeleniumDriver.getDriver()
+        .findElement(By.xpath("//button[@data-cy='" + buttonType + "']"));
+  }
+
+  public static By profilePropertiesPage() {
+    return By.xpath("//h5[contains(text(), 'Create a profile')]");
+  }
+
+  public static By locateProfileTitle(String profileName) {
+    return By.xpath("//div[@title='" + profileName + "']");
+  }
+
+  @FindBy(how = How.XPATH, using = "//a[contains(text(),\"Switch to\")]")
+  public static WebElement switchToCreatedNamespace;
+
+  @FindBy(how = How.XPATH, using = "//a[contains(text(),'Go to homepage')]")
+  public static WebElement goToHomepage;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-cy='wizard-result-icon-close-btn']")
+  public static WebElement closeAddNameSpaceWindow;
+
+  @FindBy(how = How.XPATH, using = "//span[@id='setpreferences-fd']")
+  public static WebElement setPreferencesfromMainMenu;
+
+  @FindBy(how = How.XPATH, using = "//span[contains(text(),'Upload Driver')]")
+  public static WebElement uploadDriverTab;
+
+  @FindBy(how = How.XPATH, using = "//*[contains(@class,'close')]")
+  public static WebElement closeUploadDriverWindow;
+
+  @FindBy(how = How.XPATH, using = "//span[contains(text(),'Link Repository')]")
+  public static WebElement linkRepositoryTab;
+
+  @FindBy(how = How.XPATH, using = "//button[@class=\"MuiButtonBase-root MuiIconButton-root\"]")
+  public static WebElement closeExternalPage;
+
+  @FindBy(how = How.XPATH, using = "//span[contains(text(),'Add Connection')]")
+  public static WebElement addConnectionTab;
+
+  @FindBy(how = How.XPATH, using = "//span[contains(text(),'Import')]")
+  public static WebElement importConnectionTab;
+
+  @FindBy(how = How.XPATH, using = "//span[contains(text(),'Create Profile')]")
+  public static WebElement createProfileInNamespaceAdmin;
+
+  @FindBy(how = How.XPATH, using = "//a[contains(text(),\"Switch to\")]")
```

Comment:
Locating elements based on text content is not a good idea, as explained in other comments.